reccmp/project/detect.py

```diff
@@ -71,213 +86,238 @@ class GhidraConfig:
+            return None
+
+        self.user_config_path = user_config_path
+        logger.debug("Using project config: %s", self.user_config_path)
+        return UserFile.from_file(self.user_config_path)
 
     @classmethod
-    def from_directory(cls, directory: Path) -> "RecCmpBuiltProject":
-        # Searching for build.yml
-        build_directory = find_filename_recursively(
-            directory=directory, filename=RECCMP_BUILD_CONFIG
-        )
-        if not build_directory:
-            raise RecCmpProjectNotFoundException(
-                f"Cannot find {RECCMP_BUILD_CONFIG} under {build_directory}"
-            )
-        build_config = build_directory / RECCMP_BUILD_CONFIG
-        logger.debug("Using build config: %s", build_config)
-
-        # Parse build.yml
-        build_data = BuildFile.from_file(build_config)
-
-        # Searching for project.yml
-        # note that Path.joinpath() will ignore the first path if the second path is absolute
-        project_directory = build_directory.joinpath(build_data.project)
-        project_config_path = project_directory / RECCMP_PROJECT_CONFIG
-        if not project_config_path.is_file():
-            raise InvalidRecCmpProjectException(
-                f"{build_config}: .project is invalid ({project_config_path} does not exist)"
-            )
-        logger.debug("Using project config: %s", project_config_path)
+    def from_directory(cls, directory: Path) -> "RecCmpProject":
+        project = cls()
 
-        # Parse project.yml
-        project_data = ProjectFile.from_file(project_config_path)
+        # Searching for build.yml
+        build_data = project.find_build_config(directory)
+
+        if build_data is not None:
+            assert project.build_config_path is not None
+            # note that Path.joinpath() will ignore the first path if the second path is absolute
+            project_search_path = project.build_config_path.joinpath(build_data.project)
+
+            # If we found the build file, we must use its project path.
+            project_data = project.find_project_config(project_search_path)
+            if project_data is None:
+                raise InvalidRecCmpProjectException(
+                    f"{project.build_config_path}: .project is invalid ({project_search_path / RECCMP_PROJECT_CONFIG} does not exist)"
+                )
+        else:
+            # No build file. Look for the project in the directory.
+            project_data = project.find_project_config(directory)
 
-        # Searching for user.yml
-        user_config = project_directory / RECCMP_USER_CONFIG
-        if not user_config.is_file():
-            raise InvalidRecCmpProjectException(
-                f"Missing {RECCMP_USER_CONFIG}. First run 'reccmp-project detect'."
-            )
-        logger.debug("Using user config: %s", user_config)
+        if project_data is None:
+            raise InvalidRecCmpProjectException(f"No project file in path: {directory}")
```

Comment:
```suggestion
            raise RecCmpProjectNotFoundException(f"No project file in path: {directory}")
```
I noticed this because it broke my local Ghidra setup. In case you don't want to accept this change, the Ghidra script needs a small change in order to work (in the `dev_config.json` code path)